src/domain/usecases/GetSurveyUseCase.ts

```diff
@@ -11,24 +15,44 @@ export class GetSurveyUseCase {
     constructor(private surveyReporsitory: SurveyRepository) {}
 
     public execute(
-        surveyType: SURVEY_FORM_TYPES,
-        parentSurveyId: string | undefined
+        surveyFormType: SURVEY_FORM_TYPES,
+        parentPPSSurveyId: string | undefined,
+        parentWardRegisterId: string | undefined
     ): FutureData<Questionnaire> {
-        const programId = getProgramId(surveyType);
-        if (parentSurveyId) {
-            return this.surveyReporsitory.getForm(programId, undefined).flatMap(q => {
-                const surveyIdSection = q.sections.find(
-                    s => s.questions.find(q => q.id === SURVEY_ID_DATAELEMENT_ID) !== undefined
+        const programId = getProgramId(surveyFormType);
+        if (parentPPSSurveyId) {
+            return this.surveyReporsitory.getForm(programId, undefined).flatMap(questionnaire => {
+                const surveyIdSection = questionnaire.sections.find(
+                    s =>
+                        s.questions.find(
+                            q =>
+                                q.id === SURVEY_ID_DATAELEMENT_ID ||
+                                q.id === SURVEY_ID_PATIENT_DATAELEMENT_ID
+                        ) !== undefined
                 );
                 if (surveyIdSection) {
                     const surveyIdDataElement = surveyIdSection.questions.find(
-                        q => q.id === SURVEY_ID_DATAELEMENT_ID
+                        q =>
+                            q.id === SURVEY_ID_DATAELEMENT_ID ||
+                            q.id === SURVEY_ID_PATIENT_DATAELEMENT_ID
                     );
                     if (surveyIdDataElement) {
-                        surveyIdDataElement.value = parentSurveyId;
+                        surveyIdDataElement.value = parentPPSSurveyId;
```

Comment:
Instead of changing directly the value, create a new one: 

```typescript
 const changedSections = questionnaire.sections.map(section => {
        if (
            section.questions.find(
                q => q.id === SURVEY_ID_DATAELEMENT_ID || q.id === SURVEY_ID_PATIENT_DATAELEMENT_ID
            ) !== undefined
        ) {
            const surveyIdSection = section;
            const surveyIdNewQuestions = surveyIdSection.questions.map(q => {
                return q.id === SURVEY_ID_DATAELEMENT_ID || q.id === SURVEY_ID_PATIENT_DATAELEMENT_ID
                    ? {
                          ...q,
                          value: parentPPSSurveyId,
                      }
                    : q;
            });
            return {
                ...surveyIdSection,
                questions: surveyIdNewQuestions,
            };
        }
        if (section.questions.find(q => q.id === WARD_ID_DATAELEMENT_ID) !== undefined) {
            const wardIdSection = section;
            const wardIdNewQuestions = wardIdSection.questions.map(q => {
                return q.id === WARD_ID_DATAELEMENT_ID
                    ? {
                          ...q,
                          value: parentWardRegisterId,
                      }
                    : q;
            });
            return {
                ...wardIdSection,
                questions: wardIdNewQuestions,
            };
        }
        return section;
    });
    return {
        ...questionnaire,
        sections: changedSections,
    };
```

Reply:
Done in https://github.com/EyeSeeTea/amr-surveys/pull/11